### PR TITLE
WIP apply project references to workspace packages

### PR DESCRIPTION
**Pull request by akosyakov**
let's see how it affects the build time

About project references: https://www.typescriptlang.org/docs/handbook/project-references.html

### CODE REVIEW

File: dev-packages/ext-scripts/package.json [hunk 1]
```diff
@@ -19,11 +19,11 @@
...
     "ext:build": "concurrently -n compile,lint -c blue,green \"theiaext compile\" \"theiaext lint\"",
-    "ext:compile": "tsc -p compile.tsconfig.json",
+    "ext:compile": "tsc -b",
     "ext:compile:clean": "rimraf lib",
-    "ext:lint": "tslint -c ../../configs/build.tslint.json --project compile.tsconfig.json",
-    "ext:watch": "tsc -w -p compile.tsconfig.json",
-    "ext:docs": "typedoc --tsconfig ../../configs/typedoc-tsconfig.json --options ../../configs/typedoc.json src",
+    "ext:lint": "tslint -c ../../configs/build.tslint.json --project tsconfig.json",
+    "ext:watch": "tsc -w -b",
+    "ext:docs": "typedoc --tsconfig tsconfig.json --options ../../configs/typedoc.json",
```
Comment (akosyakov) [inline dev-packages/ext-scripts/package.json]: `src` is missing

**Comment by akosyakov** [comment]
This PR does not make any difference for full build. The real bottleneck is tslint, not tsc.

Also cross references between packages work only in one direction, that is one can navigate to declaration, but cannot find usages in other packages. Maybe related to old typescript version.

**Comment by marechal-p** [comment]
I ended up doing the same thing with a script to automatically compile the references and came to the same conclusion that linitng is slow. But compilation is fast! And watching works well too, so I will also look into replacing tslint with eslint and see how faster we can get this.

**Comment by marechal-p** [comment]
Here's my branch so far https://github.com/eclipse-theia/theia/tree/mp/tsc-b

Heavily experimenting, hence why I did not open a PR yet.

**Comment by akosyakov** [comment]
see https://github.com/eclipse-theia/theia/pull/6897 instead

**Pull request closed without merging**